design/one-pager-class-selection.md

```diff
@@ -0,0 +1,371 @@
+# Simple Resource Class Selection
+* Owner: Nic Cope (@negz)
+* Reviewers: Crossplane Maintainers
+* Status: DRAFT
+
+## Background
+
+In Crossplane 0.3 we introduced support for "provider specific" (aka
+"non-portable", aka "[strongly typed]") resource classes. During the development
+cycle [we identified] that if a claim must explicitly reference a provider
+specific resource class in order to enable dynamic provisioning, then the claim
+itself is now provider specific and no longer portable across providers. We
+solved this issue by introducing [portable resource classes].
+
+A portable resource class is effectively an indirection to a non-portable
+resource class. Portable resource classes can be set as the default for a
+particular namespace, and will be used by any resource claims of their
+corresponding kind that do not specify a resource class for dynamic
+provisioning, or a managed resource for static provisioning. This can be thought
+of as "publishing" non-portable resource classes (which may exist in a distinct
+namespace used to logically group infrastructure) to other namespaces, where
+applications may use them to satisfy claims for their infrastructure
+requirements.
+
+This pattern is flexible and powerful, but Crossplane maintainers and [community
+members] have observed that it's verbose; three distinct Kubernetes resources (a
+claim, portable class, and non-portable class) must exist for dynamic
+provisioning to occur. Community members have also [provided feedback] that
+having to create portable resource classes for every new application namespace
+can be onerous.
+
+## Terminology
+
+Most terminology is covered by the [Crossplane glossary]. Key terms used in this
+document include:
+
+* Crossplane uses a _provider_ to satisfy infrastructure needs. Google Cloud
```

Comment:
In this case I'm talking about "provider" in the general sense, rather than as a the CR kind named `Provider`. I think it's useful to define here because we talk about portability across providers a lot in this design, which is my general take on the glossary - surface some terms that are important to the specific design in question and defer to the glossary for the rest.